compiler/frontend/beam/src/main/java/org/apache/nemo/compiler/frontend/beam/transform/GroupByKeyAndWindowDoFnTransform.java

```diff
@@ -52,15 +53,11 @@
   private transient InMemoryStateInternalsFactory inMemoryStateInternalsFactory;
   private Watermark prevOutputWatermark;
   private final Map<K, Watermark> keyAndWatermarkHoldMap;
+  private final WindowingStrategy windowingStrategy;
+  private Watermark inputWatermark;
```

Comment:
Remove this variable and reuse `inMemoryTimerInternalsFactory.context#currentInputWatermarkTime`?